int32 time
float32 probability
bool activate
----
bool done